ArduCopter/mode_systemid.cpp

```diff
@@ -81,14 +81,53 @@ bool ModeSystemId::init(bool ignore_checks)
     }
 
     // if landed and the mode we're switching from does not have manual throttle and the throttle stick is too high
-    if (motors->armed() && copter.ap.land_complete && !copter.flightmode->has_manual_throttle()) {
+    if (motors->armed() && copter.ap.land_complete && (!copter.flightmode->has_manual_throttle() || copter.flightmode->mode_number() == Mode::Number::LOITER)) {
         return false;
     }
 
+    if ((AxisType)axis.get() != AxisType::DISTURB_POS_LAT && (AxisType)axis.get() != AxisType::DISTURB_POS_LONG 
+         && (AxisType)axis.get() != AxisType::DISTURB_VEL_LAT && (AxisType)axis.get() != AxisType::DISTURB_VEL_LONG
+         && (AxisType)axis.get() != AxisType::INPUT_LOITER_LAT && (AxisType)axis.get() != AxisType::INPUT_LOITER_LONG) {
+
+        // System ID is being done on the attitude control loops
+            
+        if (!copter.flightmode->has_manual_throttle()) {
+            gcs().send_text(MAV_SEVERITY_WARNING, "Axis requires manual throttle");
+            return false;
+        }
+
 #if FRAME_CONFIG == HELI_FRAME
-    copter.input_manager.set_use_stab_col(true);
+        copter.input_manager.set_use_stab_col(true);
```

Comment:
Its not really related to this PR, but it looks to me like we can get left on the acro stab col when not in acro.....

Reply:
Maybe it only matters that its right in acro, stabilize and sys ID?

Reply:
@IamPete1 This only applies to acro_heli and stabilize_heli flight modes.  It is specified for the init of each of these modes.  Beings that SysID for axes 1-13 uses stabilize mode, this is set to true for heli.